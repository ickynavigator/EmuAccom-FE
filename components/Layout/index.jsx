import React from "react";
import Footer from "../Footer";
import Header from "../Header";

export const Layout = ({ children }) => {
  // TODO FIX THIS RETURN ISSUE
  console.log("Layout");

  return (
    <div id="mainApplicationWrapper">
      <Header />
      {children}
      <Footer />
    </div>
  );
};

export default Layout;
