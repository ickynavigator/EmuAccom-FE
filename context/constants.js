// TYPES
export const LOGIN_USER = "LOGIN_USER";
export const LOGOUT_USER = "LOGOUT_USER";
export const AUTHENTICATE_USER = "AUTHENTICATE_USER";
export const LOGIN_MANAGER = "LOGIN_MANAGER";
export const LOGOUT_MANAGER = "LOGOUT_MANAGER";
export const AUTHENTICATE_MANAGER = "AUTHENTICATE_MANAGER";
export const UPDATE_USER = "UPDATE_USER";
