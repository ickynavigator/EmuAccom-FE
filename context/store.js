import { useLocalStorage } from "@mantine/hooks";
import React, { createContext, useEffect, useReducer } from "react";
import combineReducers from "react-combine-reducers";
import { initialState as userState, userReducer } from "./user/reducer";

const appState = {};
const store = createContext(appState);
const { Provider } = store;

const StateProvider = ({ children }) => {
  const [reducer, initialState] = combineReducers({
    user: [userReducer, userState],
  });

  // Fetches the state from the browser local storage
  const [persistedState, setPersistedState] = useLocalStorage({
    key: "state",
    defaultValue: initialState,
  });

  const [state, dispatch] = useReducer(reducer, persistedState);

  useEffect(() => {
    // Updates the local storage state when the user makes changes
    setPersistedState(state);
  }, [setPersistedState, state]);

  return (
    // eslint-disable-next-line react/jsx-filename-extension
    <Provider value={{ state, dispatch }}>{children}</Provider>
  );
};

export { store, StateProvider };
