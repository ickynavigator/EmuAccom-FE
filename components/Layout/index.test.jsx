import { ColorSchemeProvider, MantineProvider } from "@mantine/core";
import { render } from "@testing-library/react";
import React from "react";
import { StateProvider } from "../../context/store";
import Layout from "./index";

describe("Layout component tests", () => {
  it("matches the Snapshot", () => {
    const { container } = render(
      <StateProvider>
        <ColorSchemeProvider>
          <MantineProvider>
            <Layout>
              <div>test child</div>
            </Layout>
          </MantineProvider>
        </ColorSchemeProvider>
      </StateProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
