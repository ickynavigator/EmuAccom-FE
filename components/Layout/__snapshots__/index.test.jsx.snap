// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component tests matches the Snapshot 1`] = `
<div>
  <div
    class="mantine-AppShell-root mantine-osix87"
  >
    <nav
      class="mantine-Header-root mantine-3zo516"
    >
      <div
        class="mantine-Group-root mantine-14eqvt3"
      >
        <a
          class="mantine-Text-root mantine-Anchor-root mantine-Group-child mantine-dpgj8d"
          href="#"
        >
          <div
            class="mantine-Text-root mantine-1cau8qt"
          >
            EmuAccom
          </div>
        </a>
        <button
          class="mantine-Burger-root mantine-MediaQuery-media mantine-Group-child mantine-yfh0ga"
          type="button"
        >
          <div
            class="mantine-1asblv6 mantine-Burger-burger"
          />
        </button>
        <div
          class="mantine-Group-root mantine-Group-child mantine-zah578"
        >
          <a
            class="mantine-Button-outline mantine-Button-root mantine-ad4kob"
            href="/login"
            type="button"
          >
            <div
              class="mantine-3xbgk5 mantine-Button-inner"
            >
              <span
                class="mantine-qo1k2 mantine-Button-label"
              >
                Login
              </span>
            </div>
          </a>
          <a
            class="mantine-Button-light mantine-Button-root mantine-5r4v9g"
            href="/signup"
            type="button"
          >
            <div
              class="mantine-3xbgk5 mantine-Button-inner"
            >
              <span
                class="mantine-qo1k2 mantine-Button-label"
              >
                Sign Up
              </span>
            </div>
          </a>
          <button
            class="mantine-ActionIcon-default mantine-ActionIcon-root mantine-Group-child mantine-x129ol"
            type="button"
          >
            <svg
              class="icon icon-tabler icon-tabler-moon-stars"
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
              />
              <path
                d="M17 4a2 2 0 0 0 2 2a2 2 0 0 0 -2 2a2 2 0 0 0 -2 -2a2 2 0 0 0 2 -2"
              />
              <path
                d="M19 11h2m-1 -1v2"
              />
            </svg>
          </button>
          <div
            class="mantine-Group-child mantine-1sv7zgx"
          >
            <a
              class="mantine-Text-root mantine-Anchor-root mantine-iwlv15"
              href="/about"
            >
              About
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="mantine-ibbm6b mantine-AppShell-body"
    >
      <nav
        class="mantine-Navbar-root mantine-Navbar-hidden mantine-19ywxkg"
      >
        <a
          class="mantine-Text-root mantine-Anchor-root mantine-iwlv15"
          href="/about"
        >
          About
        </a>
      </nav>
      <main
        class="mantine-1uearfw mantine-AppShell-main"
      >
        <div>
          test child
        </div>
      </main>
    </div>
  </div>
</div>
`;
